### PR TITLE
owner_test (ticdc): fix unstable case TestHandleJobsDontBlock

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve? 
fix unstable case TestHandleJobsDontBlock

https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_leak_test/detail/cdc_ghpr_leak_test/12669/pipeline/
```bash
FAIL: owner_test.go:370: ownerSuite.TestHandleJobsDontBlock

[2021-11-24T02:47:07.376Z] 

[2021-11-24T02:47:07.376Z] owner_test.go:434:

[2021-11-24T02:47:07.376Z]     c.Assert(err, check.IsNil)

[2021-11-24T02:47:07.376Z] ... value *errors.withStack = context canceled ("context canceled")
```

### What is changed and how it works?
1. add a logLimiter field for Owner to limit log output instead of use `time.sleep()`, the latter will interferes this ut case.
2. Make sure `owner.tick` will be called after statusProvider.GetAllChangeFeedInfo(ctx1) be called, otherwise the request will not be handle, so the `ctx1` will timeout.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/owner/owner_test.go [hunk 1]
```diff
@@ -405,37 +407,45 @@ func (s *ownerSuite) TestHandleJobsDontBlock(c *check.C) {
...
 	go func() {
-		infos, err := statusProvider.GetAllChangeFeedInfo(ctx1)
-		c.Assert(err, check.IsNil)
-		c.Assert(infos[changefeedID], check.NotNil)
-		c.Assert(infos[changefeedID1], check.IsNil)
+		infos, errIn = statusProvider.GetAllChangeFeedInfo(ctx1)
 		wg.Done()
 	}()
+
+	time.Sleep(1 * time.Second)
```
Comment (liuzix) [inline cdc/owner/owner_test.go:443]: Why is it necessary to wait for a second? Is there an alternative?

**Review by liuzix — COMMENTED** [review]
Can you summarize the problem in the PR description?

File: cdc/owner/owner_test.go [hunk 2]
```diff
@@ -405,37 +406,53 @@ func (s *ownerSuite) TestHandleJobsDontBlock(c *check.C) {
...
-		c.Assert(infos[changefeedID], check.NotNil)
-		c.Assert(infos[changefeedID1], check.IsNil)
-		wg.Done()
+		infos, errIn = statusProvider.GetAllChangeFeedInfo(ctx1)
+		done <- struct{}{}
 	}()
-	_, err = owner.Tick(ctx, state)
-	c.Assert(err, check.IsNil)
+
+	ticker := time.NewTicker(100 * time.Millisecond)
```
Comment (liuzix) [inline cdc/owner/owner_test.go:441]:
```suggestion
	ticker := time.NewTicker(20 * time.Millisecond)
```
This can be shorter

File: cdc/owner/owner_test.go [hunk 3]
```diff
@@ -405,37 +406,53 @@ func (s *ownerSuite) TestHandleJobsDontBlock(c *check.C) {
...
+		case <-ctx1.Done():
+			break workloop
+		case <-ticker.C:
+			_, err = owner.Tick(ctx, state)
+			c.Assert(err, check.IsNil)
+		case <-done:
+			break workloop
+		}
+	}
+	ticker.Stop()
```
Comment (liuzix) [inline cdc/owner/owner_test.go:454]: Please put `ticker.Stop()` in a defer statement.

**Review by liuzix — COMMENTED** [review]
rest LGTM

File: cdc/owner/owner_test.go (hunk 3 above)
Comment (asddongmen) [reply cdc/owner/owner_test.go:454]: ok

File: cdc/owner/owner_test.go [hunk 4]
```diff
@@ -405,37 +406,53 @@ func (s *ownerSuite) TestHandleJobsDontBlock(c *check.C) {
...
-	_, err = owner.Tick(ctx, state)
-	c.Assert(err, check.IsNil)
+
+	ticker := time.NewTicker(20 * time.Millisecond)
+	defer ticker.Stop()
+workloop:
+	for {
+		select {
+		case <-ctx1.Done():
+			break workloop
```
Comment (liuzix) [inline cdc/owner/owner_test.go:447]: Can you fail the test here?

**Review by liuzix — APPROVED** [review]

File: cdc/owner/owner_test.go [hunk 5]
```diff
@@ -405,37 +406,53 @@ func (s *ownerSuite) TestHandleJobsDontBlock(c *check.C) {
...
-		wg.Done()
+		infos, errIn = statusProvider.GetAllChangeFeedInfo(ctx1)
+		done <- struct{}{}
 	}()
-	_, err = owner.Tick(ctx, state)
-	c.Assert(err, check.IsNil)
+
+	ticker := time.NewTicker(20 * time.Millisecond)
+	defer ticker.Stop()
+workloop:
```
Comment (hi-rustin) [inline cdc/owner/owner_test.go:443]:
```suggestion
WorkoLop:
```

See: https://go.dev/doc/effective_go#switch

**Review by hi-rustin — COMMENTED** [review]
rest LGTM

File: cdc/owner/owner_test.go (hunk 4 above)
Comment (asddongmen) [reply cdc/owner/owner_test.go:447]: done

**Review by hi-rustin — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

File: cdc/owner/owner.go [hunk 6]
```diff
@@ -92,6 +93,7 @@ func NewOwner(pdClient pd.Client) *Owner {
 		gcManager:     gc.NewManager(pdClient),
 		lastTickTime:  time.Now(),
 		newChangefeed: newChangefeed,
+		logLimiter:    rate.NewLimiter(1, 1),
```
Comment (liuzix) [inline cdc/owner/owner.go:96]: Can you add the rate as a constant?

File: cdc/owner/owner.go [hunk 7]
```diff
@@ -127,7 +129,6 @@ func (o *Owner) Tick(stdCtx context.Context, rawState orchestrator.ReactorState)
 
 	if !o.clusterVersionConsistent(state.Captures) {
 		// sleep one second to avoid printing too much log
```
Comment (liuzix) [inline cdc/owner/owner.go:131]:
```suggestion

```

File: cdc/owner/owner.go (hunk 6 above)
Comment (liuzix) [reply cdc/owner/owner.go:96]: Can you comment on what logs `logLimiter` limits?

Comment (asddongmen) [reply cdc/owner/owner.go]: done

File: cdc/owner/owner.go [hunk 8]
```diff
@@ -46,6 +47,10 @@ const (
 	ownerJobTypeQuery
 )
 
+// versionInconsistentLogRate represents the rate of log output when there are
+// captures version is different with the owner version in the cluster
```
Comment (liuzix) [inline cdc/owner/owner.go:51]:
```suggestion
// captures with versions different from that of the owner
```

File: cdc/owner/owner.go [hunk 9]
```diff
@@ -280,7 +284,9 @@ func (o *Owner) clusterVersionConsistent(captures map[model.CaptureID]*model.Cap
 	myVersion := version.ReleaseVersion
 	for _, capture := range captures {
 		if myVersion != capture.Version {
-			log.Warn("the capture version is different with the owner", zap.Reflect("capture", capture), zap.String("my-version", myVersion))
+			if o.logLimiter.Allow() {
+				log.Warn("the capture version is different with the owner", zap.Reflect("capture", capture), zap.String("my-version", myVersion))
```
Comment (liuzix) [inline cdc/owner/owner.go:288]:
```suggestion
				log.Warn("the capture version is different with the owner", zap.Reflect("capture", capture), zap.String("owner-version", myVersion))
```
